### PR TITLE
Fix price discovery setup

### PR DESCRIPTION
**Pull request by claudiulataretu**

### CODE REVIEW

File: dex/pair/src/lib.rs [hunk 1]
```diff
@@ -146,6 +146,8 @@ pub trait Pair<ContractReader>:
         self.execute_output_payments(&context);
         self.emit_add_liquidity_event(&context);
 
+        self.initial_liquidity_adder().clear();
```
Comment (sasurobert) [inline dex/pair/src/lib.rs:149]: why do you need this? you just want to clear the storage ?

File: price-discovery/src/lib.rs [hunk 2]
```diff
@@ -50,7 +45,6 @@ pub trait PriceDiscovery:
             "Start epoch must be before end epoch"
         );
 
-        self.dex_sc_address().set(&dex_sc_address);
         self.launched_token_id().set(&launched_token_id);
```
Comment (sasurobert) [inline price-discovery/src/lib.rs:48]: do we already have an only owner function which can set the dex sc address ?

File: dex/pair/src/lib.rs (hunk 1 above)
Comment (sasurobert) [reply dex/pair/src/lib.rs:149]: I see - this needs to be empty on normal addLiquidity

File: price-discovery/src/lib.rs (hunk 2 above)
Comment (claudiulataretu) [reply price-discovery/src/lib.rs:48]: I added an only_owner function for this set with this PR

File: dex/pair/src/lib.rs (hunk 1 above)
Comment (claudiulataretu) [reply dex/pair/src/lib.rs:149]: There is a check in the normal `addLiquidity` endpoint to not have a price discovery address set in order to add liquidity

**Review by catalinnnn — APPROVED** [review]

**Pull request merged**